docs/csharp/language-reference/operators/patterns.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Patterns - C# reference"
+description: "Learn about the patterns supported by C# switch and is expressions and C# switch statement - C# reference"
+ms.date: 02/15/2021
+helpviewer_keywords: 
+  - "pattern matching [C#]"
+---
+# Patterns (C# reference)
+
+Intro.
+
+## Declaration and type patterns
+
+The *declaration pattern* has the following form:
```

Comment:
I've started with calling it "type pattern" as the current docs do, but then realized that the spec and, more importantly, compiler error messages use "type pattern" for the pattern that was added in C# 9 only. So, using "declaration pattern" as the spec does.